### PR TITLE
Roles refactor

### PR DESCRIPTION
**Pull request by LorranSutter**
In the current version we just use the `DEFAULT_ADMIN_ROLE` as a filter to call any restricted function, from minting devices to update modules. So much concentrated power doesn't seem very safe. We are trying to specialize each role and give it only the the necessary address, KMS, foundation or developers.

Essentially we now have:
- `DEFAULT_ADMIN_ROLE` -> add, remove, update modules and grant roles
- `ADMIN_ROLE` -> high level changes like setting proxy addresses, setting controllers, adding attributes...
- minter, burning, claiming, pairing, unpairing and setting infos roles roles for each of the node modules
- minter, burning and transferer roles for each of the NFTs (it was already there before)
- dev roles for the `DevAdmin` modules

In [deployArgs.ts](https://github.com/DIMO-Network/dimo-identity/blob/refactor/PRO-1667/identity-roles/scripts/data/deployArgs.ts) and [Roles.sol](https://github.com/DIMO-Network/dimo-identity/blob/refactor/PRO-1667/identity-roles/contracts/shared/Roles.sol) I added all roles we use

### CODE REVIEW

**Comment by elffjs** [comment]
Here or in the issue, could you write a bit about what we’re trying to accomplish? A diff this big touching something so sensitive requires some motivation, I feel.

**Comment by LorranSutter** [comment]
> Here or in the issue, could you write a bit about what we’re trying to accomplish? A diff this big touching something so sensitive requires some motivation, I feel.

Description updated

**Comment by elffjs** [comment]
Ah. I was wondering about the distinction between `DEFAULT_` and not.

File: contracts/shared/Roles.sol [hunk 1]
```diff
@@ -1,5 +1,46 @@
 //SPDX-License-Identifier: BUSL-1.1
 pragma solidity ^0.8.13;
 
+// Admin roles
 bytes32 constant DEFAULT_ADMIN_ROLE = 0x00;
-bytes32 constant MINTER_ROLE = keccak256("Minter");
+bytes32 constant ADMIN_ROLE = keccak256("ADMIN_ROLE");
```
Comment (elffjs) [inline contracts/shared/Roles.sol:6]: Can we write a little bit about the difference between these two here?

Comment (LorranSutter) [reply contracts/shared/Roles.sol:6]: Essentially  `DEFAULT_ADMIN_ROLE` can grant and revoke any role, while `ADMIN_ROLE` just have controll over high level functions, like setting a controller or whitelisting an attribute.

Maybe should call it `MANAGER_ROLE` or something else? Admin is kinda of standard I think

**Comment by elffjs** [comment]
One thing that's kinda missing from the summary is the "oracle".

File: contracts/NFTs/Base/NftBaseUpgradeable.sol [hunk 2]
```diff
@@ -21,6 +21,7 @@ abstract contract NftBaseUpgradeable is
     using CountersUpgradeable for CountersUpgradeable.Counter;
 
     CountersUpgradeable.Counter private _tokenIdCounter;
+    bytes32 public constant ADMIN_ROLE = keccak256("ADMIN_ROLE");
```
Comment (elffjs) [inline contracts/NFTs/Base/NftBaseUpgradeable.sol:24]: We don't import this from `Roles.sol`? Why?

**Comment by elffjs** [comment]
We should probably sit down and figure out who has what roles right now. Yeesh.

File: contracts/NFTs/Base/NftBaseUpgradeable.sol (hunk 2 above)
Comment (LorranSutter) [reply contracts/NFTs/Base/NftBaseUpgradeable.sol:24]: I use the `Roles.sol` only for the `DIMORegistry`. The NFTs are kind of the part.

**Pull request merged**